Small changes NtupleTools README and submit_job.py.

kdlong opened the pull request:
Made a few small changes to submit_job.py (moved two distinct ways of reading datasets into separate functions, with comments), removed "export TERMCAP=screen\n", which isn't necessary unless someone is using screen (is anyone?), and added an option to specify an output file for the script to be written to.

Also made small changes to the README in NtupleTools/test

kdlong committed:
Small changes to submit_job.py for readability

Moved argparsing to its own function. Added command line option for
specifiying an output file for output to be stored in rather than being
directed to stdout.

kdlong committed:
Small changes to submit_job.py for readability

Moved argparsing to its own function. Added command line option for
specifiying an output file for output to be stored in rather than being
directed to stdout.

kdlong committed:
Small changes to submit_job.py for readability

Moved argparsing to its own function. Added command line option for
specifiying an output file for output to be stored in rather than being
directed to stdout.

kdlong committed:
Merge branch 'miniAOD_dev' of github.com:kdlong/FinalStateAnalysis into miniAOD_dev

kdlong committed:
Improved description of output_file option

kdlong committed:
Merge commit '6f63d3fe65152eb1c9cb8abf5a76608a10a83c84' into miniAOD_dev

kdlong committed:
Learning lessons about pushing to github too soon

And looking like a total github noob in the process.

kdlong committed:
Separated das and datadef reading into separate functions

Split options to read samples from DAS and or from FSA datadefs into
separate functions. Wrote discriptions of each and improved script
discription.

kdlong committed:
Small changes to NtupleTools readme

Added information to NtupleTools/test README and made small edits to
comments in submit_jobs.py

dntaylor commented:
I use screen!

kdlong commented:
But, is there any reason to have that in by default?

kdlong commented:
My understanding is that it makes the output look better in screen, is that right? My thinking was that stuff like this can be confusing and doesn't serve a purpose for most people.

dntaylor commented:
No idea. I don't really pay attention. I just use it when I'm at my home to avoid failures during submission if my internet craps out.

dntaylor commented:
Actually, the datadefs stuff is obsolete since miniaod. Really, it can be deleted.

kdlong commented:
Even better.

dntaylor commented:
Add a commit where you just remove all the datadef parts and I'll check it tomorrow.

gomber commented:
As far as I understand screen is used when you are running something interactively and doesn't want
it to be lost due to internet connection, so I do following :

> > screen
> > run your script
> > ctrl+a d (which detach your screen)

at this point you can close the terminal, close your laptop…

when you want it back again

reopen the terminal and do

> > screen -r

---

From: Kenneth Long [notifications@github.com]
Sent: 27 February 2015 22:11
To: uwcms/FinalStateAnalysis
Subject: Re: [FinalStateAnalysis] Small changes NtupleTools README and submit_job.py. (#395)

Even better.

—
Reply to this email directly or view it on GitHubhttps://github.com/uwcms/FinalStateAnalysis/pull/395#issuecomment-76472397.

kdlong commented:
Sounds good.

dntaylor commented:
Kenneth, while you are doing all these updates/documentation, can you start adding it to a MiniAOD page on the wiki? https://github.com/uwcms/FinalStateAnalysis/wiki

kdlong commented:
I think using screen is a very good idea, I use tmux for the same purpose. But, as I understand it, this screen-specific change to an environment variable only affects the appearance of the script when run from screen (someone can check me on this). I don't think it's necessary and I prefer to have the script stay as simple as possible with less stuff in it the average user doesn't need and doesn't understand.

kdlong commented:
Thanks Devin, I'll try to make some edits there as well.

kdlong committed:
Removing datadefs data lookup, which is deprecated

Removed datadefs option, which is obsolete for miniAOD and AAA. Also
changed xrootd to go through cmsxrootd.fnal.gov, which is now
recommended.

kdlong committed:
Moved compiler flag setting to environment.sh

As it is necessary to have these compiler flags set, it seems more
straightfoward to me to have them set by default in the environment
script. This makes the setup commands slightly simpler and is less to
remember when recompiling. Devin recommends adding these commands to
your bashrc, which is reasonable, but I see no disadvantage in them
being set here as well.

dntaylor commented on Utilities/scripts/submit_job.py:
```diff
@@ -25,8 +34,287 @@
+        script_content += 'mkdir -p %s\n' % os.path.dirname(dag_dir)
+        script_content += ' '.join(command) + '\n'
+    return script_content
+def datasets_from_datadefs(args):
```
@kdlong 
Do you want to delete this out as part of the removal of datadefs?

kdlong committed:
Actually deleted all of datadefs lookup this time
